graylog2-web-interface/src/components/event-definitions/replay-search/EventInfoBar.tsx

```diff
@@ -0,0 +1,268 @@
+  user-select: none;
+  gap: 5px;
+`;
+
+const Item = styled.div`
+  display: flex;
+  gap: 5px;
+  align-items: flex-end;
+`;
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 5px;
+`;
+
+const Row = styled.div`
+  display: flex;
+  gap: 8px;
+  flex-wrap: wrap;
+`;
+const useHighlightingRules = () => useAppSelector(selectHighlightingRules);
+
+const EventInfoBar = () => {
+  useHighlightValuesForEventDefinition();
+  const dispatch = useAppDispatch();
+  const [open, setOpen] = useState<boolean>(true);
+  const allNotifications = useStore(EventNotificationsStore, ({ all }) => {
+    return all.reduce((res, cur) => {
+      res[cur.id] = cur;
+
+      return res;
+    }, {});
+  });
+  const { eventData, eventDefinition, aggregations, isEventDefinition, isEvent, isAlert } = useAlertAndEventDefinitionData();
+
+  const toggleOpen = (e) => {
+    e.stopPropagation();
+    setOpen((cur) => !cur);
+  };
+
+  const searchWithin = extractDurationAndUnit(eventDefinition.config.search_within_ms, TIME_UNITS);
+  const executeEvery = extractDurationAndUnit(eventDefinition.config.execute_every_ms, TIME_UNITS);
+
+  const notificationList = useMemo(() => {
+    return eventDefinition.notifications.reduce((res, cur) => {
+      if (allNotifications[cur.notification_id]) {
+        res.push((allNotifications[cur.notification_id]));
+      }
+
+      return res;
+    }, []);
+  }, [eventDefinition, allNotifications]);
+  const aggregationsMap = useMemo(() => new Map(aggregations.map((agg) => [`${agg.fnSeries}${agg.expr}${agg.value}`, agg])), [aggregations]);
+  const isEDUpdatedAfterEvent = !isEventDefinition && moment(eventDefinition.updated_at).diff(eventData.timestamp) > 0;
+  const highlightingRules = useHighlightingRules();
+  const highlightedAggregations = useMemo<Map<string, HighlightingRule>>(() => {
+    const initial = new Map<string, HighlightingRule>(aggregations.map(({ fnSeries, value, expr }) => [`${fnSeries}${expr}${value}`, undefined]));
+
+    return highlightingRules.reduce((acc, rule) => {
+      const { field, value, condition } = rule;
+      const expr = conditionToExprMapper?.[condition];
+      let result = acc;
+
+      if (expr) {
+        const key = `${field}${expr}${value}`;
+
+        if (acc.has(key)) {
+          result = result.set(key, rule);
+        }
+      }
+
+      return result;
+    }, initial);
+  }, [aggregations, highlightingRules]);
+
+  const changeColor = useCallback(({ rule, newColor, condition }) => {
+    if (rule) {
+      dispatch(updateHighlightingRule(rule, { color: StaticColor.create(newColor) }));
+    } else {
+      const { value, fnSeries, expr } = aggregationsMap.get(condition);
+
+      dispatch(createHighlightingRules([
+        {
+          value,
+          field: fnSeries,
+          color: randomColor(),
+          condition: exprToConditionMapper[expr],
+        },
+      ]));
+    }
+  }, [aggregationsMap, dispatch]);
+
+  const items = [
```

Comment:
This component is a little bit hard to read, maybe we can declutter the code by creating a custom hook for the items, like `useEventAttributes`. If an attribute requires a lot of code we can move its logic for `content` into a separate component.

In every case I would use a more meaningful name for `items` and memorize it.